src/dataframe/insertion.jl

```diff
@@ -297,10 +297,14 @@ function _append_or_prepend!(df1::DataFrame, df2::AbstractDataFrame; cols::Symbo
                 deleteat!(col, 1:length(col) - nrow1)
             end
         end
-        @error "Error adding value to column :$(_names(df1)[current_col]). " *
-               "Maybe it was forgotten to require column " *
-               "element type promotion, which can be done " *
-               "by passing the promote=true keyword argument."
+        if promote
+            @error "Error adding value to column :$(_names(df1)[current_col])."
+        else
+            @error "Error adding value to column :$(_names(df1)[current_col]). " *
+                "Maybe it was forgotten to require column " *
+                "element type promotion, which can be done " *
+                "by passing the promote=true keyword argument."
```

Comment:
While we're at it, maybe it would be enough to say "Maybe you forgot passing `promote=true`?"?

Reply:
OK - changed.